blues/reporters.py

```diff
@@ -0,0 +1,381 @@
+        ----------
+        simulation : simtk.openmm.app.Simulation
+            The Simulation to generate a report for
+        state : simtk.openmm.State
+            The current state of the simulation
+        """
+        if not self._is_intialized:
+            self._initialize(simulation)
+            self._is_intialized = True
+
+        self._checkForErrors(simulation, state)
+
+        args = ()
+        kwargs = {}
+        nsteps = simulation.integrator.getGlobalVariableByName('nsteps')
+        if self._coordinates:
+            coordinates = state.getPositions(asNumpy=True)[self._atomSlice]
+            coordinates = coordinates.value_in_unit(getattr(units, self._traj_file.distance_unit))
+            if not self._all_coordinates:
+                if simulation.currentStep not in [1, nsteps]:
+                    coordinates = np.zeros(coordinates.shape)
+            args = (coordinates,)
+        if self._time:
+            kwargs['time'] = state.getTime()
+        if self._cell:
+            vectors = state.getPeriodicBoxVectors(asNumpy=True)
+            vectors = vectors.value_in_unit(getattr(units, self._traj_file.distance_unit))
+            a, b, c, alpha, beta, gamma = unitcell.box_vectors_to_lengths_and_angles(*vectors)
+            kwargs['cell_lengths'] = np.array([a, b, c])
+            kwargs['cell_angles'] = np.array([alpha, beta, gamma])
+        if self._potentialEnergy:
+            kwargs['potentialEnergy'] = state.getPotentialEnergy()
+        if self._kineticEnergy:
+            kwargs['kineticEnergy'] = state.getKineticEnergy()
+        if self._temperature:
+            kwargs['temperature'] = 2*state.getKineticEnergy()/(self._dof*units.MOLAR_GAS_CONSTANT_R)
+        if self._velocities:
+            kwargs['velocities'] = state.getVelocities(asNumpy=True)[self._atomSlice, :]
+
+        #add a portion like this to store things other than the protocol work
+        if self._protocolWork:
+            protocol_work = simulation.integrator.get_protocol_work(dimensionless=True)
+            kwargs['protocolWork'] = np.array([protocol_work])
+        if self._alchemicalLambda:
+            kwargs['alchemicalLambda'] = np.array([simulation.integrator.getGlobalVariableByName('lambda')])
+        if self._title:
+            kwargs['title'] = self._title
+        if self._parameters:
+            kwargs['parameters'] = self._parameters
+        if self._environment:
+            kwargs['environment'] = self._environment
+
+        self._traj_file.write(*args, **kwargs)
+        # flush the file to disk. it might not be necessary to do this every
+        # report, but this is the most proactive solution. We don't want to
+        # accumulate a lot of data in memory only to find out, at the very
+        # end of the run, that there wasn't enough space on disk to hold the
+        # data.
+        if hasattr(self._traj_file, 'flush'):
+            self._traj_file.flush()
+
+def _check_mode(m, modes):
```

Comment:
In retrospect I probably should have just imported _check_mode directly (to prevent code duplication). You can replace the function def with an import statement:
`from mdtraj.formats.hdf5 import _check_mode`